pkg/ovsdb/transact.go

```diff
@@ -591,17 +494,28 @@ func (txn *Transaction) AddSchema(databaseSchema *libovsdb.DatabaseSchema) {
 func (txn *Transaction) Commit() (int64, error) {
 	var err error
 
-	/* verify that select is not intermixed with other operations */
+	/* verify that "select" is not intermixed with write operations */
 	hasSelect := false
-	hasOther := false
+	hasWrite := false
+Loop:
 	for _, ovsOp := range txn.request.Operations {
-		if ovsOp.Op == OP_SELECT {
+		switch ovsOp.Op {
+		case OP_SELECT:
 			hasSelect = true
-		} else {
-			hasOther = true
+			if hasSelect && hasWrite {
+				break Loop
```

Comment:
hmm if u already set hasSelect to true why u are checking it and same below for hasWrite ?

Reply:
just wrote the same test for the both switch cases, but you're right, fixed.